enhancements/storage/csi-certification.md

```diff
@@ -0,0 +1,120 @@
+
+### Goals
+
+The goal is to allow a storage vendors to be able to run a test that validates their CSI driver for use with OpenShift as a step in the process to certify their operators on the OpenShift Operator Hub.
+
+### Non-Goals
+
+* Validate CSI driver operators, this proposal targets CSI drivers only.
+* Troubleshoot or debug CSI drivers. This is the responsibility of the storage vendor.
+
+## Proposal
+
+We propose that in order to be certified, the CSI driver should meet the following requirements:
+
+* [Red Hat Universal Base Image (UBI)](https://developers.redhat.com/products/rhel/ubi/) should be used as base image for the driver.
+* The driver should use the CSI sidecars shipped by Red Hat:
+  * [csi-external-provisioner](https://github.com/openshift/csi-external-provisioner)
+  * [csi-external-attacher](https://github.com/openshift/csi-external-attacher)
+  * [csi-external-snapshotter](https://github.com/openshift/csi-external-snapshotter)
+  * [csi-external-resizer](https://github.com/openshift/csi-external-resizer)
+  * [csi-node-driver-registrar](https://github.com/openshift/csi-node-driver-registrar)
+* The driver should pass all tests contained in the `openshift/csi` test suite executed by the `openshift-tests` utility.
+  * The cluster should not have custom settings aiming the tests to pass, like SELinux disabled.
+  * The CSI driver should be installed in the cluster. This should ideally be performed via an operator.
+  * A manifest file should be created for the CSI driver.
+	* This file should describe the supported features, so that the `openshift-tests` utility knows which tests should run.
+	* Here is the [upstream documentation](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/external/README.md) that describe its format.
+  * Set the location of the manifest file through the environment variable `TEST_CSI_FRIVER_FILES` and run the CSI test suite.
+	* Example: `TEST_CSI_DRIVER_FILES=csi_driver_manifest.yaml openshift-tests run openshift/csi`
+  * It should talk a while for all tests to be executed.
+	* At the end, all tests should have passed.
+
+### User Stories [optional]
+
+#### Story 1
+
+As a storage vendor, I want to make my CSI driver available to OCP users so that they can use my storage backend.
+
+#### Story 2
+
+As OCS user, I want to storage my application's data to a 3rd party storage backend and have confidence that it'll work reliably.
+
+### Implementation Details/Notes/Constraints [optional]
+
+No technical work needs to be done in the `openshift-tests` utility, as it already contains the `openshift/csi` test suite. However, since this utility is not currently shipped in a user-friendly way, we do need a strategy for that.
```

Comment:
It is shipped as part of "tests" image in our release payload

```
$ oc adm release info registry.svc.ci.openshift.org/origin/release:4.3.0-0.okd-2019-12-17-235205
...
tests                                          sha256:eab4e9d1e612b2cee27b0a26ef56feeeb061b7e5fdfc6b829f0178601d2f46a8
```

We should promote the image to be supported and guarantee that it won't change (i.e. any change won't break any customer / driver vendor scripts).

An example how to run the tests using the image would be helpful (i.e. `podman run -v <kubeconfig> -v <manifest> -e TEST_CSI_DRIVER_FILES=<path to manifest> <image> openshift-tests <arguments>`)

Reply:
Added some notes on how to run the tests through the image